### PR TITLE
Support for AR4 min start interval and fix of dataTypeMappingRefs in NvBlock

### PR DESCRIPTION
**Pull request by kallemooo**
Added parsing and creation support for AR4 runnable min start interval.
Fixed bug with dataTypeMappingRefs in NvBlock.

### CODE REVIEW

File: autosar/behavior.py [hunk 1]
```diff
@@ -691,9 +691,11 @@ def createRunnable(self, name, portAccess=None, symbol=None, concurrent=False, e
         * exclusiveAreas: List of strings containing which exclusive areas this runnable will access.
           Note: For mode ports you will at best get read access. If you want to set new modes use modeSwitchPoints.
         * modeSwitchPoint: List of strings containing port names that this runnable will explicitly use for setting modes.
+        * minStartInterval: Specifies the time in seconds by which two consecutive starts of an ExecutableEntity are guaranteed to be separated.
```
Comment (cogu) [inline autosar/behavior.py]: I know the ARXML actually uses seconds for times. However when I designed the class autosar.behavior.TimingEvent I switched to using milliseconds in the Python API as this was more convenient for the users.
With the introduction of the minStartInterval in runnables there is now an inconsistency in the API. One part is using seconds and another part is using milliseconds.
Does it make sense to change minStartInterval to also use milliseconds for consistency?

File: autosar/writer/behavior_writer.py [hunk 2]
```diff
@@ -114,7 +114,8 @@ def _writeRunnableXML(self,runnable):
                 lines.append(self.indent('<CAN-ENTER-EXCLUSIVE-AREA-REF DEST="%s">%s</CAN-ENTER-EXCLUSIVE-AREA-REF>'%(exclusiveArea.tag(self.version),exclusiveArea.ref),2))
             lines.append(self.indent('</CAN-ENTER-EXCLUSIVE-AREA-REFS>',1))
         if self.version >= 4.0:
-            lines.append(self.indent('<MINIMUM-START-INTERVAL>%d</MINIMUM-START-INTERVAL>'%(int(runnable.minStartInterval)),1))
+            if runnable.minStartInterval is not None:
```
Comment (cogu) [inline autosar/writer/behavior_writer.py]: Check out the writer for autosar.behavior.TimingEvent how I divided by 1000.0 just before writing the XML

File: autosar/behavior.py (hunk 1 above)
Comment (kallemooo) [reply autosar/behavior.py]: I did not look so much at that, my problem where that I needed to remove the parameter in some cases as it is not supported for all SWC types.

But changing to follow autosar.behavior.TimingEvent sounds like a good solution.

File: autosar/writer/behavior_writer.py (hunk 2 above)
Comment (kallemooo) [reply autosar/writer/behavior_writer.py]: Will do

File: autosar/behavior.py [hunk 3]
```diff
@@ -680,7 +680,7 @@ def find(self,ref):
                 return foundElem
         return None
 
-    def createRunnable(self, name, portAccess=None, symbol=None, concurrent=False, exclusiveAreas=None, modeSwitchPoint = None, adminData=None):
+    def createRunnable(self, name, portAccess=None, symbol=None, concurrent=False, exclusiveAreas=None, modeSwitchPoint = None, minStartInterval = 0, adminData=None):
```
Comment (cogu) [inline autosar/behavior.py]: Well done. I will consider if it makes sense to let minStartInterval be None by default and. (Will possibly do that change later).

**Pull request merged**